packages/block-library/src/heading/edit.js

```diff
@@ -81,7 +83,9 @@ function HeadingEdit( {
 				<RichText
 					ref={ ref }
 					identifier="content"
-					tagName={ Block[ tagName ] }
+					tagName={
+						Platform.OS === 'web' ? Block[ tagName ] : tagName
```

Comment:
I think `Block` component should have an equivalent in native. This is going to become a central component to write blocks. Ideally all new blocks use it. cc @ellatrix

Reply:
Yes, ideally native has a `Block` component too. I'm not sure what it will take there. I don't know enough about the native side, but happy to collaborate with someone.

Reply:
I'm working on to have an equivalent ( more a mock at this stage) on this PR: https://github.com/WordPress/gutenberg/pull/20772

Will update this PR when the above is merged in.